utils/db_utils.py

```diff
@@ -424,100 +430,119 @@ def query_trajectories(start_date: str, end_date: str, tz: str):
+                        first_trip_ts = ts.get_first_value_for_field(
+                            key='analysis/confirmed_trip',
+                            field='data.end_ts',
+                            sort_order=pymongo.ASCENDING
+                        )
+                        if first_trip_ts != -1:
+                            user['first_trip'] = arrow.get(first_trip_ts).format(time_format)
+        
+                        last_trip_ts = ts.get_first_value_for_field(
+                            key='analysis/confirmed_trip',
+                            field='data.end_ts',
+                            sort_order=pymongo.DESCENDING
+                        )
+                        if last_trip_ts != -1:
+                            user['last_trip'] = arrow.get(last_trip_ts).format(time_format)
+        
+                        last_call_ts = ts.get_first_value_for_field(
+                            key='stats/server_api_time',
+                            field='data.ts',
+                            sort_order=pymongo.DESCENDING
+                        )
+                        if last_call_ts != -1:
+                            user['last_call'] = arrow.get(last_call_ts).format(time_format)
+                    
                 esdsq.store_dashboard_time(
-                    "admin/db_utils/add_user_stats/get_first_trip_timestamp",
-                    stage4_timer
+                    "admin/db_utils/add_user_stats/process_user",
+                    process_user_timer
                 )
-
-                # Stage 5: Get last trip timestamp
-                with ect.Timer() as stage5_timer:
-                    end_ts = ts.get_first_value_for_field(
-                        key='analysis/confirmed_trip',
-                        field='data.end_ts',
-                        sort_order=pymongo.DESCENDING
-                    )
-                    if end_ts != -1:
-                        user['last_trip'] = arrow.get(end_ts).format(time_format)
+                return user
+        esdsq.store_dashboard_time(
+            "admin/db_utils/add_user_stats/define_process_user",
+            stage1_timer
+        )
+        
+        # Stage 2: Define batch_process
+        with ect.Timer() as stage2_timer:
+            def batch_process(users_batch):
+                with ect.Timer() as batch_process_timer:
+                    with ThreadPoolExecutor() as executor:  # Adjust max_workers based on CPU cores
+                        futures = [executor.submit(process_user, user) for user in users_batch]
+                        processed_batch = [future.result() for future in as_completed(futures)]
                 esdsq.store_dashboard_time(
-                    "admin/db_utils/add_user_stats/get_last_trip_timestamp",
-                    stage5_timer
+                    "admin/db_utils/add_user_stats/batch_process",
+                    batch_process_timer
```

Comment:
Again, renaming metrics will make it challenging to compare the before and the after.

Reply:
I named it differently because the functionality is different; what do you suggest -- keep the old name or just track total time?

Reply:
old name